### PR TITLE
Fix: typing

### PR DESCRIPTION
**Pull request by kopax**
improve all typings, added necessary export such as SwiperControlsProps used when overriding component

### CODE REVIEW

**Comment by kopax** [comment]
Anyone ?

File: index.d.ts [hunk 1]
```diff
@@ -73,7 +73,7 @@ export interface BadgeProps {
      *
      * @default View (if onPress then TouchableOpacity)
      */
-    Component?: React.ComponentClass;
+    Component?: React.ComponentType<T>;
```
Comment (jarvisluong) [inline index.d.ts:76]: If <T> is used a type param, you need to add it as a type param of BadgeProps as well

File: index.d.ts [hunk 2]
```diff
@@ -174,12 +174,12 @@ interface SwiperControlsProps {
     /**
      * Custom prev button component
      */
-    PrevComponent?: React.ComponentClass;
+    PrevComponent?: React.ComponentType<T>;
```
Comment (jarvisluong) [inline index.d.ts:177]: Same with these type generics

**Review by jarvisluong — CHANGES_REQUESTED** [review]

File: index.d.ts (hunk 2 above)
Comment (kopax) [reply index.d.ts:177]: I can't commit those change from the UI, can you?

File: index.d.ts (hunk 1 above)
Comment (kopax) [reply index.d.ts:76]: I can't commit those change from the UI, can you?

Comment (jarvisluong) [reply index.d.ts:76]: Not sure what you meant, what I was trying to say was that adding <T> like this is not enough. The <T> should come from BadgeProps<T> as well.

**Comment by jarvisluong** [comment]
This PR seems to be stale now. Will close. Please feel free to reopen if needed

**Pull request closed without merging**